app/Models/Device.php

```diff
@@ -210,7 +210,7 @@ public function displayName(): string
     {
         $hostname_is_ip = IP::isValid($this->hostname);
 
-        return SimpleTemplate::parse($this->display ?: \LibreNMS\Config::get('device_display_default', '{{ $hostname }}'), [
+        return SimpleTemplate::parse(htmlentities($this->display) ?: \LibreNMS\Config::get('device_display_default', '{{ $hostname }}'), [
```

Comment:
I don't think this is the correct place for sanitation.

Reply:
same with format_hostname

Reply:
Suggestions welcome, I tried fixing it more lower down the code base so I didn't touch these functions but found it still didn't resolve the issue :/